### ISSUE TITLE
ResponseLogger prints raw request on cancellation

### ISSUE THREAD

**Issue by sebbert** [http4s/http4s#3482]
On this line of code, we log the result of calling `toString` on the request:
https://github.com/http4s/http4s/blob/a7e2fc108de12f7b111d688d33288c20f6a5a085/server/src/main/scala/org/http4s/server/middleware/ResponseLogger.scala#L70

This is dangerous, as it may end up logging sensitive information, ignoring the `logHeaders`, `logBody`, and `redactHeadersWhen` options.

In our project, we've solved this by using `Logger.logMessage` to log the request in this case, re-using the same options as for responses and introducing a constraint on the request type, `A <: Message[F]`.

However, this may not be a good solution for getting this patched upstream, since it technically breaks the API (introducing a new type constraint), and re-uses the response logging configuration for requests.

**Comment by rossabaker** [comment]
Ouch. Yeah, we need to get this fixed ASAP.

**Comment by rossabaker** [comment]
So if I'm reading the code right, the default will redact these headers, and your problem is that you have other sensitive headers?

```scala
val SensitiveHeaders = Set(
  "Authorization".ci,
  "Cookie".ci,
  "Set-Cookie".ci
)
```

This is tricky.  Constraining `A` to be a `Message[F]` defeats the polymorphism of the middleware.  We could pattern match on `A` and if it's a request, do something safer.  But then we'd need to handle `ContextRequest`.  And who's to say that people haven't invented their own `A` with a `Request`?

The only right answer here might be to be much more careful about what we put in `.toString`, since we can't pass the sensitive headers filter reliably in this context.

**Comment by rossabaker** [comment]
Also, much less gravely, while we're at it, `canceled` is one `l`. :smile:

**Comment by sebbert** [comment]
> So if I'm reading the code right, the default will redact these headers, and your problem is that you have other sensitive headers?

Well, we actually disable header `logHeaders` altogether, but we do have additional sensitive headers, yes.

**Comment by sebbert** [comment]
This might be a stupid question, but would it be possible to move this behavior to `RequestLogger`?

**Comment by rossabaker** [comment]
Fixed by #3488.
